python/housinginsights/sources/cama.py

```diff
@@ -0,0 +1,208 @@
+    Use public methods to retrieve data.
+    """
+
+    BASEURL = 'https://opendata.arcgis.com/datasets'
+    def __init__(self):
+        super().__init__(CamaApiConn.BASEURL)
+
+    def get_data(self):
+        """
+        Grabs data from CAMA. Individual CAMA property retrieves zone_type data
+        from MAR api. Count number of housing units and bedroom units per zone.
+        Return the count data (in dictionary form) to be processed into csv
+        by get_csv() method.
+        """
+
+        mar_api = MarApiConn()
+        result = self.get(urlpath='/c5fb3fbe4c694a59a6eef7bf5f8bc49a_25.geojson', params=None)
+
+        if result.status_code != 200:
+            err = "An error occurred during request: status {0}"
+            raise Exception(err.format(result.status_code))
+        cama_data = result.json()
+
+        """
+        Example of: anc_count = [OrderedDict([('zone_type', 'anc'), ('zone', 'ANC 2B'), 
+                                ('housing_unit_count', 10), ('bedroom_unit_count', 10)], etc)]
+        """
+        zone_types = ['anc', 'census_tract', 'neighborhood_cluster', 'ward', 'zip']
+        anc_count = []
+        census_count = []
+        cluster_count = []
+        ward_count = []
+        zipcode_count = []
+
+
+        """
+        Take each CAMA property data and retrieve the MAR data.
+        """
+        """
+        Certain square values have four digits + a letter. (ex. 8888E)
+        Square would be the first four digits and suffix would be the letter.
+        SSL sometimes comes as 8 digit string without spacing in the middle.
+        """
+        """
+        CAMA data includes bldgs under construction. CAMA's data includes AYB of 2018
+        as of June 2017. We eliminate all data points that are under construction and
+        don't provide any housing units and bedrm at this time.
+        """
+        for row in cama_data['features']:
+            try:
+                current_year = int(datetime.date.today().strftime('%Y'))
+                #Skipping none values for units under construction
+                if row['properties']['AYB'] is not None and int(row['properties']['AYB']) > current_year:
+                    continue
+
+                objectid = row['properties']['OBJECTID']
+                if len(row['properties']['SSL']) == 8:
+                    square = row['properties']['SSL'][:4]
+                    lot = row['properties']['SSL'][4:]
+                else:
+                    square, lot = row['properties']['SSL'].split()
+                suffix = ' '
+                if len(square) > 4:
+                    square = square[:4]
+                    suffix = square[-1]
+
+                mar_return = mar_api.get_data(square, lot, suffix)
+
+                ''' Count the housing units and bedrooms '''
+                num_units = 0
+                if row['properties']['NUM_UNITS']: num_units = row['properties']['NUM_UNITS']
+                if num_units == 0:
```

Comment:
In the CAMA data, buildings with 0 num_units (the data for number of housing units) falls under number of cases. There are about 177 data points with 0 housing units. 

- First set of CAMA data that has 0 num_units and 0 bedrooms. However, they have associated MAR data points. Also their use codes are residential. Some examples are,
    (SSL: 1264 0808) // according to Integrated Tax System API: land use code 023 - Residential Flats-Less than 5
    (SSL: 1256 0078) // land use code 012 - Residential-Detached-Single-Family
The assumption was that these buildings are used for the purpose of residential because they are taxed as such. Therefor there should be at least 1 bedroom and therefore it counts as 1 housing unit. There should be less than 10 data points that meet this criteria, so even if the assumption is off, it shouldn't really be that big of a deal in a data point with 100,000+ data points. 

- Second set is certain data points for single family homes (and the most numerous data points with 0 num_units). The have multiple bedrooms but they show up as 0 num_units even though it should be 1. They also have MAR data point when you call it on the MAR API. An example is (SSL: 0069 0805). CAMA data shows 0 num_units and 3 bedrooms. MAR API has appropriate data. When looking at Integrated Tax System API you see building code 011 for Residential-Row-Single-Family. These housing units should be counted as 1 housing unit instead of 0, as they contain bedrooms and serve as a housing unit. There are about 154 data points out of 177.

- Third set of data point with 0 housing units are data point with 0 housing units and 0 bedroom but has no MAR data associated with them. An example is (SSL: 0514 0871). They are excluded from our data points as there is no way to figure out which zone they belong in. There are less than 20 data points are in CAMA. 

Summary: There are data points in CAMA with 0 housing units, but most are single family or equivalent housing stock and should be counted as 1 housing unit instead of 0.